bitnami/nats/templates/secrets.yaml

```diff
@@ -11,7 +11,107 @@ metadata:
   {{- if .Values.commonAnnotations }}
   annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
   {{- end }}
-data:
+stringData:
+  {{- $authPwd := include "nats.password" . -}}
+  {{- $clusterAuthPwd := include "nats.clusterPassword" . -}}
+  {{ if .Values.configuration }}
   {{ .Values.natsFilename }}.conf: |-
-    {{- include "common.tplvalues.render" (dict "value" .Values.configuration "context" $) | b64enc | nindent 4 }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.configuration "context" $ ) | nindent 4 }}
+  {{- else -}}
+  {{- if .Values.auth.enabled }}
+  {{- if .Values.auth.user }}
+  user-password: {{ $authPwd | quote }}
+  {{- else if .Values.auth.token }}
+  token: {{ .Values.auth.token | quote }}
+  {{- end }}
+  {{- end }}
+  {{- if .Values.cluster.auth.enabled }}
+  {{- if .Values.cluster.auth.user }}
+  cluster-password: {{ $clusterAuthPwd | quote }}
+  {{- else if .Values.cluster.auth.token }}
+  token: {{ .Values.cluster.auth.token | quote }}
+  {{- end }}
+  {{- end }}
```

Comment:
The key `token` is duplicated. I would like to suggest the following unique keys: 
- `password`
- `token`
- `cluster-password`
- `cluster-token`

Reply:
In addition, please move these conditionals out of the `{{ if .Values.configuration }}` conditional.